src/AtomsMCMoves.jl

```diff
@@ -0,0 +1,368 @@
+                    at::AtomWalker, 
+                    lj::LJParameters, 
+                    step_size::Float64, 
+                    emax::typeof(0.0u"eV")
+                    )
+    n_accept = 0
+    accept_this_walker = false
+    for i_mc_step in 1:n_steps
+        config = at.configuration
+        frozen = at.num_frozen_part
+        e_shift = at.energy_frozen_part
+        i_at = rand((frozen+1):length(config))
+        pos::SVector{3, typeof(0.0u"Å")} = position(config, i_at)
+        orig_pos = deepcopy(pos)
+        pos = single_atom_random_walk!(pos, step_size)
+        pos = periodic_boundary_wrap!(pos, config)
+        config.position[i_at] = pos
+        energy = interaction_energy(config, lj; frozen=frozen) + e_shift
+        if energy >= emax
+            # reject the move, revert to original position
+            config.position[i_at] = orig_pos
+        else
+            at.energy = energy
+            # accept the move
+            n_accept += 1
+            accept_this_walker = true
+        end
+    end
+    return accept_this_walker, n_accept/n_steps, at
+end
+
+
+"""
+    single_atom_demon_walk!(at::AtomWalker, lj::LJParameters, step_size::Float64;
+                            e_demon=0.0u"eV",
+                            demon_energy_threshold=Inf*u"eV",
+                            demon_gain_threshold=Inf*u"eV")
+
+Perform a single atom demon walk.
+
+Arguments:
+- `at::AtomWalker`: The walker to perform the demon walk on.
+- `lj::LJParameters`: The LJ parameters.
+- `step_size::Float64`: The step size for the random walk.
+
+Keyword Arguments:
+- `e_demon=0.0u"eV"`: The energy of the demon.
+- `demon_energy_threshold=Inf*u"eV"`: The energy threshold for the demon.
+- `demon_gain_threshold=Inf*u"eV"`: The energy gain threshold for the demon during each move.
+
+Returns:
+- `accept::Bool`: Whether the move is accepted or rejected.
+- `at::AtomWalker`: The updated atom walker object.
+- `e_demon::Float64`: The updated energy of the demon.
+"""
+function single_atom_demon_walk!(
+                        at::AtomWalker, 
+                        lj::LJParameters, 
+                        step_size::Float64;
+                        e_demon=0.0u"eV",
+                        demon_energy_threshold=Inf*u"eV",
+                        demon_gain_threshold=Inf*u"eV",
+                        )
+    accept = false
+    orig_energy = at.energy
+    config = at.configuration
+    frozen = at.num_frozen_part
+    e_shift = at.energy_frozen_part
+    i_at = rand((frozen+1):length(config))
+    pos = position(config, i_at)
+    orig_pos = deepcopy(pos)
+    pos = single_atom_random_walk!(pos, step_size)
+    pos = periodic_boundary_wrap!(pos, config)
+    config.position[i_at] = pos
+    new_energy = interaction_energy(config, lj; frozen=frozen) + e_shift
+    ΔE = new_energy - orig_energy
+    if -demon_gain_threshold <= ΔE <= 0.0u"eV" && e_demon-ΔE < demon_energy_threshold
+        e_demon -= ΔE
```

Comment:
Is `+= -∆E` less cognitively dissonant than `-= ∆E`? It took me a second to parse lines 199-212.